File: recruitment-service/src/main/java/pe/albrugroup/recruitment_service/entity/response/OfertaLaboralResponse.java
package pe.albrugroup.recruitment_service.entity.response;

import lombok.*;
import pe.albrugroup.recruitment_service.entity.enums.EstadoOferta;
import pe.albrugroup.recruitment_service.entity.enums.Modalidad;
import pe.albrugroup.recruitment_service.entity.enums.Negocio;
import pe.albrugroup.recruitment_service.entity.enums.PuestoObjetivo;
import pe.albrugroup.recruitment_service.entity.enums.TurnoHorario;

import java.time.Instant;
import java.time.LocalDate;
import java.util.List;

@Builder @Getter @Setter
@AllArgsConstructor @NoArgsConstructor
public class OfertaLaboralResponse {

    private Long id;
    private String codigo;
    private Long idSolicitante;
    private Negocio negocio;
    private PuestoObjetivo puestoObjetivo;
    private Modalidad modalidad;
    private TurnoHorario horario;
    private Integer cantidadInicial;
    private LocalDate plazoInicial;
    private EstadoOferta estado;
    private Instant createdAt;
    private List<OfertaAmpliacionResponse> ampliaciones;
}

File: frontend/src/app/shared/models/recruitment/oferta-laboral-response.ts
import { OfertaAmpliacionResponse } from './oferta-ampliacion-response';

export interface OfertaLaboralResponse {
  id: number;
  codigo: string;
  idSolicitante: number;
  negocio: string;
  puestoObjetivo: string;
  modalidad: string;
  horario: string;
  cantidadInicial: number;
  plazoInicial: string;
  estado: string;
  createdAt: string;
  ampliaciones: OfertaAmpliacionResponse[];
}
